build/config/compiler/BUILD.gn

```diff
@@ -195,14 +195,17 @@ config("disabled_warnings") {
   cflags = [
     "-Wno-deprecated-declarations",
     "-Wno-missing-field-initializers",
-    "-Wno-unknown-warning-option",
```

Comment:
Why wasn't this being an issue before?

Reply:
No idea .... I just got a failure in CI and added this to see if CI passes. We had green CI before.

https://github.com/project-chip/connectedhomeip/actions/runs/5192198591/jobs/9361029995

The error is:

```
In file included from ../../src/app/CommandHandler.h:33,
439
                 from ../../src/app/util/generic-callbacks.h:25,
440
                 from ../../src/app/util/util.cpp:25:
441
../../src/app/ConcreteCommandPath.h:32:40: error: expected ')' before ',' token
442
   32 |     ConcreteCommandPath(EndpointId aEndpointId, ClusterId aClusterId, CommandId aCommandId) :
443
      |                            ~           ^
444
      |                                        )
445
cc1plus: error: unrecognized command line option '-Wno-unknown-warning-option' [-Werror]
446
cc1plus: all warnings being treated as errors
```

The error seems to be expected ')' and the log seems off though, so I do not quite understand it.
It may not be the real failure but maybe we never saw it because we do not read passing logs. Will check some older passing runs.

Reply:
It does look like a correct fix since `g++ --help=warnings | grep unknown-warning` comes up empty for me (but others like `g++ --help=warnings | grep missing-field` are not)

Reply:
I do not know what happened with taht failure. There is no , on line 32 in that file, but there is one on line 33, so somehow the compiler report is off-by-one. my delta in compile option seems to be a drive-by change rather than actually addressing the real failure, however likely still correct (given that gcc does not undestand that option).

Now that CI step passes in this PR.

Reply:
So to be clear: Either we need that -Wno... for clang, and then we need it for clang period, which is not accurately reflected by `is_clang`.

Or we don't need it, and then we should not have it.

Reply:
will wait for darwin to actually pick up the runs and see.
If we need it, I will add it back (since is_clang will be false on darwin even if gcc/g++ redirects to clang)
If we do not need it, I will remove it completely.

Reply:
It seems darwin failed now with

```
error: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Werror,-Wunknown-warning-option]
```

so I guess for darwin we need to tell it to ignore unknown options because we pass unknown options. Probably could be untangled with platform settings, but that seems too much yak shaving. Reverted the change.